src/descriptors/power_system_structs.json

```diff
@@ -2450,6 +2450,55 @@
       ],
       "supertype": "Reserve{T}"
     },
+    {
+      "struct_name": "StaticReserveGroup",
+      "docstring": "Data Structure for a group reserve product for system simulations.",
+      "parametric": "ReserveDirection",
+      "fields": [
+        {
+          "null_value": "init",
+          "name": "name",
+          "accessor_module": "InfrastructureSystems",
+          "data_type": "String"
+        },
+        {
+          "null_value": "false",
+          "name": "available",
+          "data_type": "Bool"
+        },
+        {
+          "name": "requirement",
+          "comment": "the static value of required reserves in system p.u.",
+          "null_value": "0.0",
+          "data_type": "Float64",
+          "valid_range": {
+            "min": 0,
+            "max": null
+          },
+          "validation_action": "error"
+        },
+        {
+          "name": "ext",
+          "data_type": "Dict{String, Any}",
+          "null_value": "Dict{String, Any}()",
+          "default": "Dict{String, Any}()"
+        },
+        {
+          "name": "contributing_services",
+          "comment": "Services that contribute for this requirement constraint",
+          "data_type": "Vector{<:Service}",
```

Comment:
This doesn't work for some reason! It doesn't create a `Vector{<:Service}` type in the generated file, so I had to modify it by hand (which breaks the generated struct test). Any idea on how to proceed?

Reply:
What does it generate?  Also, are you sure that you don't want `Vector{Service}`?

Reply:
> Does it print something like Vector{&lt;:Service})? 

Yes!

> Are you sure that you don't want Vector{Service}? 

I have tried that but, for some reason, it doesn't accept subtypes of `Service` in the `set_contributing_services` function.

Reply:
I proposed a fix to the auto-generation issue here:  https://github.com/NREL-SIIP/InfrastructureSystems.jl/pull/112/files.

That said, you could fix the issue you saw in your test by explicitly declaring an array of type Service. 

```
array = Vector{Service}()
push!(array, service)
```

Either way works for me.

Reply:
My bad, that's true! I fixed it with your suggestion!